nav2_safety_nodes/src/nav2_safety_node.cpp

```diff
@@ -0,0 +1,250 @@
+#include <chrono>
+#include <memory>
+#include <vector>
+#include <string>
+#include <utility>
+
+#include "nav2_safety_nodes/nav2_safety_node.hpp"
+
+using namespace std::chrono_literals;
+
+namespace nav2_safety_nodes
+{
+
+SafetyZone::SafetyZone()
+: nav2_util::LifecycleNode("SafetyZone", "", false)
+{
+  logger_ = get_logger();
+  RCLCPP_INFO(logger_, "Creating Safety Polygon");
+
+  // Vector of string for multiple LaserScan topics
+  const std::vector<std::string> scan{};
+  // pass polygon parameters at string
+  declare_parameter("safety_polygon", std::string("[]"));
+  declare_parameter("zone_action", rclcpp::ParameterValue(0.0));
+  declare_parameter("zone_priority", rclcpp::ParameterValue(1));
+  declare_parameter("zone_num_pts", rclcpp::ParameterValue(1));
+  declare_parameter("base_frame", std::string("base_link"));
+  declare_parameter("tf_tolerance", rclcpp::ParameterValue(0.01));
+  declare_parameter("scan_topics", scan);
+  declare_parameter("speed_limit_topic", rclcpp::ParameterValue("speed_limit"));
+}
+
+SafetyZone::~SafetyZone()
+{
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_configure(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(logger_, "Configuring");
+  getParameters();
+  initTransforms();
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_activate(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(logger_, "Activating");
+  initPubSub();
+  safety_polygon_pub_->on_activate();
+  point_cloud_pub_->on_activate();
+  speed_limit_pub_->on_activate();
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_deactivate(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(logger_, "Deactivating");
+  safety_polygon_pub_->on_deactivate();
+  point_cloud_pub_->on_deactivate();
+  speed_limit_pub_->on_deactivate();
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_cleanup(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(logger_, "Cleaning up");
+  safety_polygon_pub_.reset();
+  point_cloud_pub_.reset();
+  speed_limit_pub_.reset();
+  timer_.reset();
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_shutdown(const rclcpp_lifecycle::State &)
+{
+  RCLCPP_INFO(logger_, "Shutting down");
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+// Get Parameters
+void
+SafetyZone::getParameters()
+{
+  RCLCPP_DEBUG(logger_, " getParameters");
```

Comment:
remove this logging, replace with an INFO message that has actual sentence structure, not just debugging artifacts.